django/contrib/auth/backends.py

```diff
@@ -96,6 +97,23 @@ def has_module_perms(self, user_obj, app_label):
                 return True
         return False
 
+    def with_perm(self, perm, is_active=True, is_superuser=True, obj=None):
+        UserModel = get_user_model()
+        try:
+            app_label, codename = perm.split('.')
+        except ValueError:
+            raise ValueError("Permission name should be in the form 'app_label.perm_name'.")
```

Comment:
I think that we should accept a `Permission` instance as well as the dot-separated string...

Reply:
Good idea. Will add support for `Permission` instances later tonight.

Reply:
Done!